kf/nfconfig_test.go

```diff
@@ -417,7 +417,511 @@ func Test_getHostInterfaces(t *testing.T) {
+				hostName:       "l3af-local-test",
+				hostInterfaces: map[string]bool{"enp0s3": true},
+				mu:             new(sync.Mutex),
+				ingressXDPBpfs: map[string]*list.List{"enp0s3": nil},
+				ingressTCBpfs:  map[string]*list.List{"enp0s3": nil},
+				egressTCBpfs:   map[string]*list.List{"enp0s3": nil},
+			},
+			arg: args{
+				hostName: "l3af-local-test",
+				iface:    "enp0s3",
+				bpfProgs: &models.BPFPrograms{
+					XDPIngress: []*models.BPFProgram{},
+					TCEgress:   []*models.BPFProgram{},
+					TCIngress:  []*models.BPFProgram{},
+				},
+			},
+			wanterr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cfg := &NFConfigs{
+				hostName:       tt.field.hostName,
+				IngressXDPBpfs: tt.field.ingressXDPBpfs,
+				IngressTCBpfs:  tt.field.ingressTCBpfs,
+				EgressTCBpfs:   tt.field.egressTCBpfs,
+				hostConfig:     tt.field.hostConfig,
+				processMon:     tt.field.processMon,
+				hostInterfaces: tt.field.hostInterfaces,
+				mu:             tt.field.mu,
+			}
+			err := cfg.AddProgramsOnInterface(tt.arg.iface, tt.arg.hostName, tt.arg.bpfProgs)
+			if (err != nil) != tt.wanterr {
+				t.Errorf("AddProgramsOnInterface: %v", err)
+			}
+		})
+	}
+}
+
+func TestAddeBPFPrograms(t *testing.T) {
+	type fields struct {
+		hostName       string
+		hostInterfaces map[string]bool
+		ingressXDPBpfs map[string]*list.List
+		ingressTCBpfs  map[string]*list.List
+		egressTCBpfs   map[string]*list.List
+		hostConfig     *config.Config
+		processMon     *pCheck
+		mu             *sync.Mutex
+		ifaces         map[string]string
+	}
+	tests := []struct {
+		name    string
+		field   fields
+		arg     []models.L3afBPFPrograms
+		wanterr bool
+	}{
+		{
+			name: "UnknowHostName",
+			field: fields{
+				hostName: "l3af-prod",
+				ifaces:   map[string]string{},
+				hostConfig: &config.Config{
+					L3afConfigStoreFileName: "/etc/l3afd/l3af-config.json",
+				},
+			},
+			arg: []models.L3afBPFPrograms{
+				{
+					HostName: "l3af-test",
+					Iface:    "enp0s3",
+					BpfPrograms: &models.BPFPrograms{
+						XDPIngress: []*models.BPFProgram{},
+						TCIngress:  []*models.BPFProgram{},
+						TCEgress:   []*models.BPFProgram{},
+					},
+				},
+			},
+			wanterr: true,
+		},
+		{
+			name: "NilInterface",
+			field: fields{
+				hostName: "l3af-local-test",
+				ifaces:   map[string]string{},
+				hostConfig: &config.Config{
+					L3afConfigStoreFileName: "/etc/l3afd/l3af-config.json",
+				},
+			},
+			arg: []models.L3afBPFPrograms{
+				{
+					HostName: "enp0s3",
+					BpfPrograms: &models.BPFPrograms{
+						XDPIngress: []*models.BPFProgram{},
+						TCIngress:  []*models.BPFProgram{},
+						TCEgress:   []*models.BPFProgram{},
+					},
+				},
+			},
+			wanterr: true,
+		},
+		{
+			name: "UnknownInterface",
+			field: fields{
+				hostName: "l3af-local-test",
+				ifaces:   map[string]string{},
+				hostConfig: &config.Config{
+					L3afConfigStoreFileName: "/etc/l3afd/l3af-config.json",
```

Comment:
Can this use a relative path so that this can work on both linux and windows?
Seems like we shouldn't need different .go files per platform.

Reply:
No this is not using relative path.
You can look #34. 
I think in future we will use some kind of database for configs.

Reply:
Is this string also ignored by the test and could be arbitrary or does the file need to actually exist?  If it's arbitrary then I suggest:
```suggestion
					L3afConfigStoreFileName: "dummy_file_name",
```

Reply:
No this file need to exist otherwise it will create file.
I don’t know where we are storing configs in windows.